pkg/store/store.go

```diff
@@ -462,3 +563,63 @@ func toNetworkingIngressV1Beta1(obj *extensions.Ingress) (*networkingv1beta1.Ing
 	out.APIVersion = networkingv1beta1.SchemeGroupVersion.String()
 	return &out, nil
 }
+
+// convUnstructuredObj is a convenience function to quickly convert any runtime.Object where the underlying type
+// is an *unstructured.Unstructured (client-go's dynamic client type) and convert that object to a runtime.Object
+// which is backed by the API type it represents. You can use the GVK of the runtime.Object to determine what type
+// you want to convert to. This function is meant so that storer implementations can optionally work with YAML files
+// for caller convenience when initializing new CacheStores objects.
+//
+// TODO: upon some searching I didn't find an analog to this over in client-go (https://github.com/kubernetes/client-go)
+//       however I could have just missed it. We should switch if we find something better, OR we should contribute
+//       this functionality upstream.
+func convUnstructuredObj(obj *runtime.Object, convertedObj runtime.Object) error {
+	_, isUnstructured := (*obj).(*unstructured.Unstructured)
+	if !isUnstructured {
+		return nil
+	}
+
+	var b []byte
+	b, err := yaml.Marshal(obj)
+	if err != nil {
+		return fmt.Errorf("failed to convert object %s to yaml: %w", (*obj).GetObjectKind().GroupVersionKind(), err)
+	}
+	if err = yaml.Unmarshal(b, convertedObj); err != nil {
+		return err
+	}
+	*obj = convertedObj
+	return nil
+}
+
+// mkObjFromGVK is used to provide a new runtime.Object given a GVK, and is particularly useful for handling
+// unstructured runtime.Objects and converted them to structured.
```

Comment:
```suggestion
// mkObjFromGVK is a factory function that returns a concrete implementation runtime.Object
// for the given GVK. Callers can then use `convert()` to convert an unstructured
// runtime.Object into a concrete one.
```

this is nitpicky, but :shrug: : let's talk about the function, not about what its callers _use it to_ do